users/helauth/parsers.py

```diff
@@ -40,11 +50,43 @@ def parse(self) -> ReservationPrefillInfo:
             home_city=self.get_user_home_city(),
         )
 
-    def get_last_name(self) -> str | None:
-        return self.data.get("lastName")
+    def parse_birthday_info(self) -> BirthdayInfo:
+        return BirthdayInfo(
+            id=self.data.get("id"),
+            birthday=self.get_birthday(),
+        )
+
+    def parse_user_profile_info(self) -> UserProfileInfo:
+        address = self.get_address()
+        return UserProfileInfo(
+            first_name=self.get_first_name(),
+            last_name=self.get_last_name(),
+            email=self.get_email(),
+            phone=self.get_phone(),
+            birthday=self.get_birthday(),
+            ssn=self.get_social_security_number(),
+            street_address=address.get("address"),
+            postal_code=address.get("postalCode"),
+            city=address.get("city"),
+            municipality_code=self.get_municipality_code(),
+            municipality_name=self.get_municipality_name(),
+            login_method=LoginMethod.PROFILE,
+            is_strong_login=True,  # Helsinki profile login is always strong
```

Comment:
Not necessarily in the future, could be password-only weak profile also in practice, if login with that is allowed at some point.
First thought was that this could be filled in from _user.id_token_ like here (whether its `loa` is `substantial`): https://github.com/City-of-Helsinki/tilavarauspalvelu-core/pull/1155/files#diff-3b2de07e9621e0695c8cd04e23191a197382943f8866bc9052a8d5352d85e6e1R71
**However**, that would simply reflect the last login method of the user in cases where profile has been created with both authentication methods (both `amr = suomi_fi` and `amr = helsinki_tunnus` i.e. password associated the same email in Helsinki-profile, where token for both gives the same `sub` claim but the latter has `loa = low` ).
**But** looks like it's possible to figure out that if `verifiedPersonalInformation` is not returned in the profile response despite being queried, then it's ultimately a weak/password-only profile.

Reply:
> But looks like it's possible to figure out that if verifiedPersonalInformation is not returned in the profile response despite being queried, then it's ultimately a weak/password-only profile.

Where did you get this information? Is it simply null in this case, or will there be an error? Currently, if the request to profile has any errors, our endpoint will fail.

Reply:
null as part of the returned profile but the response also contains an error: "No permission to read verified personal information."
But come to think of it, I'll have to check if the response would be different when the profile has no suomi_fi association at all and could not be returned.

Reply:
Counting on an error to tell us whether a profile is a certain kind seems like a bad idea. Their GraphQL documentation also states the following:

> **verifiedPersonalInformation**: **VerifiedPersonalInformationNode**
Personal information that has been verified to be true. Can result into `PERMISSION_DENIED_ERROR` if the requester has no required privileges to access this information.

This tells me that the requester's permissions are insufficient for querying this information, not that the requested user does no have it.

Reply:
What is the use of `is_strong_login` on this endpoint anyway? Is it to indicate whether the user has strong login associated with their Helsinki profile account at all? In this case, it should be Helsinki profile providing that information to us. I added a comment on this to make a change once weak auth is implemented there, but before that, I don't think we need to do anything more here.

Reply:
Yes to indicate whether a strong login is associated with the profile.
Could be simply based on self.data.get("verifiedPersonalInformation") is not None?

Reply:
Added this to the parser.